igor-web/src/main/groovy/com/netflix/spinnaker/igor/health/PollingMonitorHealth.groovy

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2014 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License')
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.igor.health
+
+import com.netflix.spinnaker.igor.jenkins.BuildMonitor
+import com.netflix.spinnaker.igor.polling.PollingMonitor
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.actuate.health.Health
+import org.springframework.boot.actuate.health.HealthIndicator
+import org.springframework.boot.actuate.health.Status
+import org.springframework.stereotype.Component
+
+/**
+ * The health status is based on whether the poller is running and how long ago it last polled. If twice the polling
+ * interval has passed since the last poll the poller is considered _down_.
+ */
+@Component
+public class PollingMonitorHealth implements HealthIndicator {
+
+    @Autowired
+    List<PollingMonitor> pollers
+
+    @Override
+    public Health health() {
+        List<Health> healths = []
```

Comment:
Basically, it checks if both the Docker and Build polling agents are running, and squashes their statuses together (see line 57).